packages/ui-source-code-editor/src/SourceCodeEditor/index.tsx

```diff
@@ -469,9 +469,9 @@ class SourceCodeEditor extends Component<SourceCodeEditorProps> {
 
     if (this.direction === 'rtl' && !rtlMoveVisually) {
       // we need to clone the original so that it doesn't get overridden
-      return merge(
-        cloneDeep(defaultKeymap),
-        rtlHorizontalArrowKeymap
+      return mergeDeep(
+        defaultKeymap as unknown as Record<string, unknown>,
+        rtlHorizontalArrowKeymap as unknown as Record<string, unknown>
```

Comment:
The original `merge` is actually does not work the way we expect, thus creating a bug in this logic, for which I created a separate bug ticket.

So I think this is fine to merge and fix the bug in another PR.

Reply:
I would still at least change it to `[...defaultKeymap, ...rtlHorizontalArrowKeymap]`, since we realized that they are arrays, and mergeDeep accepts objects :) And then leave a comment to fix the broken logic